apis/python/src/tiledb/vector_search/object_api/object_index.py

```diff
@@ -0,0 +1,390 @@
+    )
+    abs_path = os.path.abspath(f"{temp_file_name}.py")
+    f = open(abs_path, "w")
+    f.write(code)
+    f.close()
+    spec = importlib.util.spec_from_file_location(temp_file_name, abs_path)
+    reader_module = importlib.util.module_from_spec(spec)
+    sys.modules[temp_file_name] = reader_module
+    spec.loader.exec_module(reader_module)
+    class_ = getattr(reader_module, class_name)
+    os.remove(abs_path)
+    return class_(**kwargs)
+
+
+def create(
+    uri: str,
+    index_type: str,
+    object_reader: ObjectReader,
+    embedding: ObjectEmbedding,
+    config: Optional[Mapping[str, Any]] = None,
+    storage_version: str = STORAGE_VERSION,
+    **kwargs,
+) -> ObjectIndex:
+    with tiledb.scope_ctx(ctx_or_config=config):
+        dimensions = embedding.dimensions()
+        vector_type = embedding.vector_type()
+        if index_type == "FLAT":
+            index = flat_index.create(
+                uri=uri,
+                dimensions=dimensions,
+                vector_type=vector_type,
+                group_exists=False,
+                config=config,
+                storage_version=storage_version,
+            )
+        elif index_type == "IVF_FLAT":
+            index = ivf_flat_index.create(
+                uri=uri,
+                dimensions=dimensions,
+                vector_type=vector_type,
+                group_exists=False,
+                config=config,
+                storage_version=storage_version,
+            )
+
+        group = tiledb.Group(uri, "w")
+        group.meta["object_reader_source_code"] = get_source_code(object_reader)
+        group.meta["object_reader_class_name"] = object_reader.__class__.__name__
+        group.meta["object_reader_kwargs"] = encode_class(object_reader.init_kwargs())
+        group.meta["embedding_source_code"] = get_source_code(embedding)
+        group.meta["embedding_class_name"] = embedding.__class__.__name__
+        group.meta["embedding_kwargs"] = encode_class(embedding.init_kwargs())
+
+        embeddings_array_name = storage_formats[index.storage_version][
+            "INPUT_VECTORS_ARRAY_NAME"
+        ]
+        filters = storage_formats[index.storage_version]["DEFAULT_ATTR_FILTERS"]
+
+        create_embeddings_partitioned_array(
+            uri, embeddings_array_name, group, vector_type, filters
+        )
+        object_metadata_array_uri = object_reader.metadata_array_uri()
+        materialize_object_metadata = False
+        if (
+            object_metadata_array_uri is None
+            and object_reader.metadata_attributes() is not None
+        ):
+            metadata_array_name = storage_formats[index.storage_version][
+                "OBJECT_METADATA_ARRAY_NAME"
+            ]
+            object_metadata_array_uri = f"{uri}/{metadata_array_name}"
+            external_ids_dim = tiledb.Dim(
+                name="external_id",
+                domain=(0, np.iinfo(np.dtype("int32")).max),
```

Comment:
Why a signed 32 bit int and not unsigned 64bit?

Reply:
fixed